File: fixtures/adapter-java/basic/src/main/java/com/example/UserService.java
package com.example;

// Everything below is resolvable inside this one file, which is exactly the
// limit of Tier 2: the analyzer records what it can see here and nothing more.
public interface Auditable {
    String describe();
}

class AuditLog {
    void record(String message) {
    }
}

public class UserService implements Auditable {
    private AuditLog log;

    public String describe() {
        return "user service";
    }

    public void handle() {
        // A call to a method defined in this file.
        describe();
    }
}

File: fixtures/adapter-csharp/basic/UserService.cs
// Everything below is resolvable inside this one file, which is exactly the
// limit of Tier 2: the analyzer records what it can see here and nothing more.
namespace Example
{
    public interface IAuditable
    {
        string Describe();
    }

    class AuditLog
    {
        public void Record(string message) { }
    }

    public class UserService : IAuditable
    {
        private AuditLog log;

        public string Describe()
        {
            return "user service";
        }

        public void Handle()
        {
            // A call to a method defined in this file.
            Describe();
        }
    }
}
